oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsSoundCard.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Oshi (https://github.com/oshi/oshi)
+ * <p>
+ * Copyright (c) 2010 - 2018 The Oshi Project Team
+ * <p>
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * <p>
+ * Maintainers:
+ * dblock[at]dblock[dot]org
+ * widdis[at]gmail[dot]com
+ * enrico.bianchi[at]gmail[dot]com
+ * <p>
+ * Contributors:
+ * https://github.com/oshi/oshi/graphs/contributors
+ */
+package oshi.hardware.platform.windows;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import oshi.hardware.common.AbstractSoundCard;
+import oshi.jna.platform.windows.WbemcliUtil;
+import oshi.util.platform.windows.WmiUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Sound Card data obtained by Wmi Queries for Windows.
+ *
+ * @author : BilalAM
+ */
+public class WindowsSoundCard extends AbstractSoundCard {
+
+    enum SoundCardName {
+        MANUFACTURER, NAME;
+    }
+
+    enum SoundCardKernel {
+        DRIVERPROVIDERNAME, DRIVERNAME, DRIVERVERSION, DEVICENAME;
+    }
+
+    private static final Logger LOG = LoggerFactory.getLogger(WindowsSoundCard.class);
+    private static WbemcliUtil.WmiQuery<SoundCardName> AUDIO_CARD_QUERY = new WbemcliUtil.WmiQuery<>("Win32_SoundDevice",SoundCardName.class);
+    private static WbemcliUtil.WmiResult<SoundCardName> AUDIO_CARD_QUERY_RESULT = WmiUtil.queryWMI(AUDIO_CARD_QUERY);
```

Comment:
Wouldn't it better to just make the AUDIO_CARD_QUERY static and final since it is not going to change throughout along with the AUDIO_CARD_QUERY_RESULT instead of using static blocks ?

Reply:
That's what I'm trying to point out -- you don't need both.  Pick one or the other.  If you persist the result as final, you don't need the query any more.   So you could declare the result as final and put a static block after it to define/run the query, then the query can go away.

Or if you want to query and generate the result on-the-fly then persist the query but not the result -- generate the result down in the getter methods when you need it.  This is how we do it in most other OSHI classes: we persist the query as static and final, and only hold on to the results temporarily.    

However, I understand that SoundCards are not hot-swappable and all this information can probably be generated one time and stored in String lists or maps, and then you can dispose of all the WMI stuff and just fetch from the maps when the getters are called.